pkg/gzip/gzip_test.go

```diff
@@ -47,6 +45,10 @@ func TestFileWithoutDir(t *testing.T) {
 }
 
 func TestSelfHardlink(t *testing.T) {
+	if runtime.GOOS == "darwin" {
+		t.Skip("bsdtar doesn't handle a selflink tarball")
```

Comment:
Wasn't it the case that different versions of GNU's tar implementation had different behaviour wrt self-hardlink? In that case, this test captures a regression in an integration style. I support that.